modules/boulder/src/main/ts/ephox/boulder/core/ValueProcessor.ts

```diff
@@ -1,262 +1,205 @@
-import { Adt, Arr, Fun, Merger, Obj, Optional, Thunk, Type } from '@ephox/katamari';
+import { Arr, Fun, Merger, Obj, Optional, Thunk, Type } from '@ephox/katamari';
 import { SimpleResult, SimpleResultType } from '../alien/SimpleResult';
-
 import * as FieldPresence from '../api/FieldPresence';
 import * as Objects from '../api/Objects';
 import { ResultCombine } from '../combine/ResultCombine';
 import * as ObjWriter from './ObjWriter';
 import * as SchemaError from './SchemaError';
+import * as ValuePresence from './ValuePresence';
 
 type SchemaError = SchemaError.SchemaError;
 
-// TODO: Handle the fact that strength shouldn't be pushed outside this project.
-export type Strength = (res: any) => any;
-export type ValueValidator = (a, strength?: Strength) => SimpleResult<string, any>;
-export type PropExtractor = (path: string[], strength: Strength, val: any) => SimpleResult<SchemaError[], any>;
-export type ValueExtractor = (label: string, prop: Processor, strength: Strength, obj: any) => SimpleResult<SchemaError[], string>;
+export type ValueValidator = (a) => SimpleResult<string, any>;
+export type PropExtractor = (path: string[], val: any) => SimpleResult<SchemaError[], any>;
+export type ValueExtractor = (label: string, prop: Processor, obj: any) => SimpleResult<SchemaError[], string>;
 export interface Processor {
-  extract: PropExtractor;
+  extractProp: PropExtractor;
```

Comment:
Was there a reason this was renamed?

Reply:
Too many things called `extract`, was driving me crazy, so I renamed a few. Happy to take name suggestions, just don't want the four different things to all be called `extract` still

Reply:
Renaming to `getProp` to shorten + further differentiate